src/main/java/seedu/task/logic/history/TaskMementos.java

```diff
@@ -23,7 +23,7 @@ public TaskMementos() {
      * @param taskMemento the task memento to add to the undo history
      */
     public void addUndoMementoAndClearRedo(TaskMemento taskMemento) {
-        this.undoMementoStack.push(taskMemento);
+        this.undoMementoStack.push(new TaskMemento(taskMemento.oldTask, taskMemento.newTask));
```

Comment:
Do you have to copy here? The `TaskMemento` should be immutable anyway and the tasks are copied in the `TaskMemento` constructor.